capstone/src/main/webapp/js/autocomplete.js

```diff
@@ -57,56 +83,45 @@ function closeAllElmntExcept(elmnt) {
  * Adds valid suggestions to list of stored suggestions
  * @param {string} validSuggestion
  */
-function keepTrackOfSuggestions(validSuggestion) {
+function keepTrackOfChosenColleges(validSuggestion) {
   const parent = document.getElementById('suggestions');
   const listElt = document.createElement('li');
-  listElt.innerHTML = validSuggestion;
+  listElt.setAttribute('id', validSuggestion.collegeId.id);
+  listElt.innerHTML = validSuggestion.collegeName;
   parent.appendChild(listElt);
 }
 
 /**
- * Current placeholder until servlet is created
- * @return {Array<string>} arr
- */
-function getListOfSuggestions() {
-  const arr = ['Hallo', 'Hello', 'Hi', 'Hiya', 'Howdy', 'Wassup'];
-  return arr;
-}
-
-/**
- * Identifies and returns relevant suggestions in the arr
- * @param {string[]} arr
+ * Creaters List Element to display a college name suggestion
+ * @param {HTMLDivElement} arrElt
  * @param {string} val
- * @return {string[]} relevantSuggestions
+ * @return {HTMLDivElement} listElmt
  */
-function getRelevantDataSuggestions(arr, val) {
-  const relevantSuggestions = [];
-  val = val.toUpperCase();
-  for (const arrElt of arr) {
-    if (arrElt.substr(0, val.length).toUpperCase() === val.toUpperCase()) {
-      relevantSuggestions.push(arrElt);
-    }
-  }
-  return relevantSuggestions;
+function createListElmt(arrElt, val) {
+  const listElmt = document.createElement('div');
+  const collegeNameUpper = arrElt.collegeName.toUpperCase();
+  const valIndex = collegeNameUpper.indexOf(val.toUpperCase());
+  listElmt.innerHTML = arrElt.collegeName.substr(0, valIndex);
+  listElmt.innerHTML += '<strong>' +
+                       arrElt.collegeName.substr(valIndex, val.length) +
+                       '</strong>';
+  listElmt.innerHTML += arrElt.collegeName.substr(val.length + valIndex);
```

Comment:
This seems like it would benefit from some unit testing :)

Reply:
done!